dsgrid/query/models.py

```diff
@@ -200,10 +205,56 @@ class ProjectQueryParamsModel(DSGBaseModel):
         description="Controls table format",
         default=TableFormatType.PIVOTED,
     )
+
+    @validator("dimension_filters")
+    def handle_dimension_filters(cls, dimension_filters):
+        for i, dimension_filter in enumerate(dimension_filters):
+            if not isinstance(dimension_filter, DimensionFilterBaseModel):
+                dimension_filters[i] = make_dimension_filter(dimension_filter)
+        return dimension_filters
+
+    @validator("per_dataset_aggregations")
+    def check_aggregations(cls, aggregations):
+        for agg in aggregations:
+            for dim_type in DimensionType:
+                field = dim_type.value
+                for item in getattr(agg.dimensions, field):
+                    # TODO: we could add support for this, but it does add some ambiguity.
+                    if item.function is not None:
+                        raise ValueError(
+                            f"function={item.function} cannot be set in ProjectQueryParamsModel"
```

Comment:
I think this is for `ProjectQueryDatasetParamsModel`, same for error msg below. Still confused why we can't have column func here.